horovod/run/mpi_run.py

```diff
@@ -88,7 +107,7 @@ def mpi_run(settings, common_intfs, env, command):
         common_intfs=','.join(common_intfs)) if common_intfs else ''
 
     # On large cluster runs (e.g. Summit), we need extra settings to work around OpenMPI issues
-    if settings.num_hosts >= 64:
+    if settings.num_hosts and settings.num_hosts >= _LARGE_CLUSTER_THRESHOLD:
         mpi_impl_flags.append('-mca plm_rsh_no_tree_spawn true')
```

Comment:
It's interesting that they introduced this change for mpi_run but not for the spark_run.  One of the reasons we override the mpi settings is to set this value to true.

Reply:
horovod.spark.run does currently not set settings.num_hosts so there will be no way to reuse this code through horovod.spark.run. Maybe we could derive num_hosts from settings.hosts, then this extra args would be set automatically by mpi_run when needed.